bundles/action/org.openhab.action.pushover/src/main/java/org/openhab/action/pushover/internal/Pushover.java

```diff
@@ -343,15 +383,47 @@ public static boolean pushover(@ParamDoc(name = "apiKey", text = "Your applicati
             }
 
             if (!StringUtils.isEmpty(sound)) {
-                addEncodedParameter(data, MESSAGE_KEY_SOUND, sound);
+                parts.add(new StringPart(MESSAGE_KEY_SOUND, sound, UTF_8_ENCODING));
             } else if (!StringUtils.isEmpty(defaultSound)) {
-                addEncodedParameter(data, MESSAGE_KEY_SOUND, defaultSound);
+                parts.add(new StringPart(MESSAGE_KEY_SOUND, defaultSound, UTF_8_ENCODING));
             }
 
-            String content = data.toString();
-            logger.debug("Executing post to {} with the following content: {}", XML_API_URL, content);
-            String response = HttpUtil.executeUrl("POST", XML_API_URL, IOUtils.toInputStream(content), CONTENT_TYPE,
-                    timeout);
+            if (!StringUtils.isEmpty(attachment)) {
+                logger.debug("Push the image attachment '{}'.", attachment);
+                try {
+                    File f = new File(attachment);
+                    FilePart fp = new FilePart(MESSAGE_KEY_ATTACHMENT, f.getName(), f);
+                    fp.setContentType("image/jpeg");
```

Comment:
And we still need a fix for the content-type.

Reply:
what is your proposal here? to pass an additional parameter?

Reply:
Maybe the `HttpUtil.guessContentTypeFromData()` method can help: https://github.com/eclipse/smarthome/blob/master/bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java#L496

Reply:
could be an option, though not my favourite one, since you need to read the whole binary file into an byte array, you can end up having memory issues
feel free, to implement it in an additional PR

Reply:
I agree. Reading the whole file is a little bit too much overhead. I am looking for a different way. [URLConnection.guessContentTypeFromName(file.getName())](https://docs.oracle.com/javase/7/docs/api/java/net/URLConnection.html#guessContentTypeFromName(java.lang.String)) could be an option, but it has a very limited set of content types that it does recognizes.

Reply:
in 99% one will be fine with the default content type
as user you have now the option to set the content type explicitly, so why should i guess the content type ?
```
sendPushoverMessage(pushoverBuilder("test message").withAttachment("/path/to/file.pdf").withContentType("application/pdf"))
```
another question is, whether pushover supports that type of files and how they are supported... its not documented, see https://pushover.net/api#attachments